packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx

```diff
@@ -31,15 +34,41 @@ export const RainbowKitCustomConnectButton = () => {
 
               if (chain.unsupported || chain.id !== configuredNetwork.id) {
                 return (
-                  <>
-                    <span className="text-xs" style={{ color: networkColor }}>
-                      {configuredNetwork.name}
-                    </span>
-                    <button className="btn btn-sm btn-error ml-2" onClick={openChainModal} type="button">
+                  <div className="dropdown dropdown-end">
+                    <button tabIndex={0} className="btn btn-error btn-sm dropdown-toggle">
                       <span>Wrong network</span>
                       <ChevronDownIcon className="h-6 w-4 ml-2 sm:ml-0" />
                     </button>
-                  </>
+                    <ul tabIndex={0} className="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-56">
+                      <li>
+                        <button
+                          className="menu-item"
+                          type="button"
+                          onClick={() => {
+                            if (switchNetwork) {
+                              switchNetwork(configuredNetwork.id);
+                            }
+                          }}
+                        >
+                          <ArrowsRightLeftIcon className="h-6 w-4 ml-2 sm:ml-0" />
+                          <span>
+                            Switch to <span style={{ color: networkColor }}>{configuredNetwork.name}</span>
+                          </span>
+                        </button>
+                      </li>
+                      <li>
+                        <button
+                          className="menu-item text-red-500"
+                          type="button"
+                          onClick={() => {
+                            disconnect();
+                          }}
```

Comment:
Interested in knowing why this happens XD @sverps 

Never seen it before, for me they should be the same thing and I'd always use `onClick={disconnect}`

Reply:
Updated with `onClick={() => disconnect()}`

Reply:
Here is the parsed error : 
![Screenshot 2023-04-20 at 3 31 18 PM](https://user-images.githubusercontent.com/80153681/233333185-2b21a005-308e-4318-9c42-dfb248056085.jpg)

The type of `disconnect` : `UseMutateFunction<void, Error, void, unknown>`  
typeof `onClick` for button : `MouseEventHandler<HTMLButtonElement>`

The place where their both types mismatch in function `parameters` : 

Actual type of `onClick` is somewhat : `(arg : MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => ...` 

and I think resolved type of `disconnect` is: `(arg : void) => ...`

so the `arg` is mismatched, 

--- 

Doing `onClick={() => discconect()}` works because I think here we have just inlined function without assigning any types to inlined function not even to  `args`  and its all infered 

If we do `onClick={(e: void) => disconnect()}` we get same error as above : 
![Screenshot 2023-04-20 at 3 46 05 PM](https://user-images.githubusercontent.com/80153681/233336699-42af63e2-4486-462c-b215-d36cfc885cac.jpg)


Lol that's my two cents of knowledge 😂...but would really love to hear from @sverps about the error and also please free to point out the wrong thing it will help fill in my knowledge gaps 🙏

Reply:
Yeah, your explanation is indeed correct.

The `disconnect` function comes from tanstack react-query and it's a mutate function, which takes the variables of the mutation as a first param.
```typescript
const disconnect: (variables: void, options?: MutateOptions<void, Error, void, unknown> | undefined) => void
```

In this case, it's a mutation without variables, hence the `void`.

If we were to place it directly in the `onClick`, it would try to pass down the `MouseEvent` as a first param, which would indeed be wrong. So typescript is there again to save the day :sweat_smile:

Reply:
Thanks guys <3